libs/agno/agno/workflow/v2/workflow.py

```diff
@@ -0,0 +1,503 @@
+            "execution_start": execution_start,
+        }
+
+        # Update agents and teams with workflow session info
+        self.update_agents_and_teams_session_info()
+
+        # Collect complete workflow run instead of individual events
+        workflow_run_responses = []
+
+        try:
+            # Execute the sequence synchronously
+            for response in sequence.execute(inputs, context):
+                # Collect all responses
+                workflow_run_responses.append(response)
+                yield response
+
+            # Store only the complete workflow run (not individual events)
+            if self.workflow_session and workflow_run_responses:
+                # Store only the final completed workflow response
+                # The workflow_completed event
+                final_response = workflow_run_responses[-1]
+                if final_response.event == WorkflowRunEvent.workflow_completed:
+                    self.workflow_session.add_run(final_response)
+
+            # Save to storage after complete execution
+            self.write_to_storage()
+
+        except Exception as e:
+            logger.error(f"Workflow execution failed: {e}")
+
+            error_response = WorkflowRunResponse(
+                content=f"Workflow execution failed: {e}",
+                event=WorkflowRunEvent.workflow_error,
+                workflow_id=self.workflow_id,
+                workflow_name=self.name,
+                sequence_name=sequence_name,
+                session_id=self.session_id,
+                run_id=self.run_id,
+            )
+
+            # Store error response
+            if self.workflow_session:
+                self.workflow_session.add_run(error_response)
+            self.write_to_storage()
+
+            yield error_response
+
+    def update_agents_and_teams_session_info(self):
+        """Update agents and teams with workflow session information"""
+        # Update all agents in sequences
+        for sequence in self.sequences:
+            for task in sequence.tasks:
+                active_executor = task._active_executor
+
+                if hasattr(active_executor, "workflow_session_id"):
+                    active_executor.workflow_session_id = self.session_id
+                if hasattr(active_executor, "workflow_id"):
+                    active_executor.workflow_id = self.workflow_id
+
+                # If it's a team, update all members
+                if hasattr(active_executor, "members"):
+                    for member in active_executor.members:
+                        if hasattr(member, "workflow_session_id"):
+                            member.workflow_session_id = self.session_id
+                        if hasattr(member, "workflow_id"):
+                            member.workflow_id = self.workflow_id
+
+    def run(self, query: str = None, sequence_name: Optional[str] = None, **kwargs) -> Iterator[WorkflowRunResponse]:
```

Comment:
please add session_id, user_id, etc. Pretty much all the stuff you can usually call run on agent/team with. Images etc as well